src/core/server.py

```
@@ -115,7 +124,19 @@ async def crawl4ai_lifespan(server: FastMCP) -> AsyncIterator[Crawl4AIContext]:
     
+    # Track active operations to prevent premature cleanup
+    active_operations = set()
+    cleanup_lock = asyncio.Lock()
+    
     try:
+        # Setup enhanced logging
+        setup_mcp_logging(log_level="INFO", enable_structured=True)
+        logger.info("🔧 Enhanced MCP logging configured")
+        
+        # Initialize operation monitoring
+        monitor = OperationMonitor()
```

Ruff (0.12.2)

Local variable `monitor` is assigned to but never used

Remove assignment to unused variable `monitor`

(F841)